process_testcase.go

```diff
@@ -253,36 +257,43 @@ func (v *Venom) runTestSteps(ctx context.Context, tc *TestCase) {
 				}
 			}
 
-			v.RunTestStep(ctx, e, tc, stepNumber, step)
+			v.setTestStepName(ts, e, step, &ranged, &rangedData)
+			if v.Verbose > 0 && !fromUserExecutor {
+				v.Print(" \t\t• %s", ts.Name)
+			}
 
+			v.RunTestStep(ctx, e, tc, ts, stepNumber, rangedIndex, step)
 			tc.testSteps = append(tc.testSteps, step)
 
 			var hasFailed bool
 			var isRequired bool
-			if len(tc.Failures) > 0 {
-				for _, f := range tc.Failures {
+			if len(ts.Failures) > 0 {
+				for _, f := range ts.Failures {
 					Warning(ctx, "%v", f)
 					isRequired = isRequired || f.AssertionRequired
 				}
 				hasFailed = true
 			}
 
-			if len(tc.Errors) > 0 {
+			if len(ts.Errors) > 0 {
 				Error(ctx, "Errors: ")
-				for _, e := range tc.Errors {
+				for _, e := range ts.Errors {
 					Error(ctx, "%v", e)
 				}
 				hasFailed = true
 			}
 
 			if hasFailed {
 				if isRequired {
-					failure := newFailure(*tc, stepNumber, "", fmt.Errorf("At least one required assertion failed, skipping remaining steps"))
-					tc.Failures = append(tc.Failures, *failure)
+					failure := newFailure(*tc, stepNumber, rangedIndex, "", fmt.Errorf("At least one required assertion failed, skipping remaining steps"))
+					ts.appendFailure(tc, *failure)
+					v.printTestStepResult(tc, ts, tsIn)
 					return
 				}
-				break
+				v.printTestStepResult(tc, ts, tsIn)
+				continue
```

Comment:
`break` replaced by `continue`, else next step weren't executed which contradicts the default behaviour